portfolio/src/main/java/com/google/sps/servlets/ScheduleServlet.java

```diff
@@ -91,4 +117,35 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
     return tasks;
   }
+
+  private static Optional<SchedulingAlgorithmType> getSchedulingAlgorithmTypeOptional(
+      String algorithmTypeString) {
+    // Here we should add a case for each new algorithm that is implemented.
+    switch (algorithmTypeString) {
+      case "SHORTEST_TASK_FIRST":
+        return Optional.of(SchedulingAlgorithmType.SHORTEST_TASK_FIRST);
+    }
+    return null;
+  }
+
+  private static Optional<BaseTaskScheduler> getAlgorithmOptional(
```

Comment:
Consider renaming to getTaskSchedulerOptional

Reply:
Done.